From 4fac94ae08f1d0f4f95d01fcbb618dee13b582c9 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 11 Jul 2022 20:30:23 +0300
Subject: [PATCH] [SW-94367] gaudi3: move disable pcie response after fuse check

In order to make sure no PCIe domain registers are configured if
PCIE_CTRL fuse value is disabled, this patch performs disabling of PCIe
error response based on PCIE_CTRL efuse value.

Change-Id: I744a10b9a8389a0587e90e17ef48cfde56b2b7ba
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 2ddc6d8..1cd6d84 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -102,17 +102,17 @@
 {
 	uint32_t pcie_ctrl_val = arcboot_rom_global.config_fuse & EFUSE_SYSCFG_PCIE_CONTROL_MASK;
 
-	/* Disable PCIe error responses (both HBW and LBW) */
-	regmap_writel(0x1, mmPCIE_RSP_OK);
-	regmap_writel(0x1, mmD0_PCIE_WRAP_PCIE_RSP_OK_1);
-
 	/*
 	 * Master die (die_id=0): pcie_ctrl 0: PCIe enable 1: PCIe disable
 	 * Slave die (die_id=1): pcie_ctrl 0: PCIe disable 1: PCIe enable
 	 */
 	if ((arcboot_rom_global.die_id && pcie_ctrl_val) ||
-		(!arcboot_rom_global.die_id && !pcie_ctrl_val))
+		(!arcboot_rom_global.die_id && !pcie_ctrl_val)) {
+		/* Disable PCIe error responses (both HBW and LBW) */
+		regmap_writel(0x1, mmPCIE_RSP_OK);
+		regmap_writel(0x1, mmD0_PCIE_WRAP_PCIE_RSP_OK_1);
 		pcie_config();
+	}
 }
 
 #define D2D_LINK_STATUS_RETRIES		100
